Phoenix/Controllers/LoginController.cs

```diff
@@ -40,67 +40,63 @@ public ActionResult Authenticate(LoginViewModel loginViewModel)
             // Get the username and password from the view model
             string username = loginViewModel.Username;
             string password = loginViewModel.Password;
-            if (password == null)
+
+            try
             {
-                // if user did not enter a password, set it to empty string, since this value must be non-null
-                password = "";
+                _ADContext = loginService.ConnectToADServer();
             }
-
-            if (!ModelState.IsValid)
+            catch (Exception e)
             {
-                loginViewModel.ErrorMessage = "Invalid model state.";
+                loginViewModel.ErrorMessage = "There was a problem connection to the server. We are sorry. Please try again later or contact the project maintainer.";
                 return View("Index", loginViewModel);
             }
-            else
+      
+            UserPrincipal userEntry;
+            try
+            {
+                userEntry = loginService.FindUser(username, _ADContext);
+            }
+            catch(UserNotFoundException ex)
+            {
+                _ADContext.Dispose();
+                loginViewModel.ErrorMessage = "Oh dear, it seems that username or password is invalid.";
+                return View("Index", loginViewModel);
+            }
+            catch(ArgumentNullException ex)
+            {
+                _ADContext.Dispose();
+                loginViewModel.ErrorMessage = "Please make sure you enter in values for the username.";
```

Comment:
Don't we NOT want to tell the user why a login didn't work? I remember discussing that. So it should always say a generic message about "username or password is incorrect" on the user side, even if on our side the reason why the login failed looks different.